autogluon_zeroshot/simulation/tabular_predictions.py

```diff
@@ -0,0 +1,233 @@
+        return list(set.intersection(*map(set, models)))
+
+    @property
+    def datasets(self) -> List[str]:
+        return list(self.pred_dict.keys())
+
+    def remove_dataset(self, dataset: str):
+        self.pred_dict.pop(dataset)
+
+    def rename_datasets(self, rename_dict: dict):
+        for key in rename_dict:
+            assert key in self.datasets
+        self.pred_dict = {rename_dict.get(k, k): v for k, v in self.pred_dict.items()}
+
+    @property
+    def folds(self) -> List[int]:
+        # todo we could assert that the same number of folds exists in all cases
+        first = next(iter(self.pred_dict.values()))
+        return list(first.keys())
+
+
+class TabularPicklePerTaskPredictions(TabularModelPredictions):
+    def __init__(self, dataset_to_models: Dict[str, List[str]], output_dir: str):
+        """
+        Stores on pickle per task and load data in a lazy fashion which allows to reduce significantly the memory
+        footprint.
+        :param dataset_to_models:
+        :param output_dir:
+        """
+        self.dataset_to_models = dataset_to_models
+        self.output_dir = Path(output_dir)
+        self.rename_dict_inv = {}
+        assert self.output_dir.is_dir()
+
+    def _score(self, dataset: str, fold: int, splits: List[str] = None, models: List[str] = None) -> List[np.array]:
+        dataset = self.rename_dict_inv.get(dataset, dataset)
+        pred_dict = self._load_dataset(dataset)
+        if models is None:
+            models = self.models_available_per_dataset(dataset, fold)
+
+        def get_split(split, models):
+            split_key = 'pred_proba_dict_test' if split == "test" else 'pred_proba_dict_val'
+            model_results = pred_dict[fold][split_key]
+            return np.array([model_results[model] for model in models])
+        return [get_split(split, models) for split in splits]
+
+    def _load_dataset(self, dataset: str) -> Dict:
+        filename = str(self.output_dir / f'{dataset}.pkl')
+        return load_pkl.load(filename)
+
+    @classmethod
+    def from_dict(cls, pred_dict: TabularPredictionsDict, output_dir: str = None):
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        pred_proba = TabularPicklePredictions.from_dict(pred_dict=pred_dict)
+        datasets = pred_proba.datasets
+        dataset_to_models = {
+            dataset: pred_proba.models_available_per_dataset(dataset)
+            for dataset in datasets
+        }
+        print(f"saving .pkl files in folder {output_dir}")
+        for dataset in datasets:
+            filename = str(output_dir / f'{dataset}.pkl')
+            print(filename)
+            save(filename, pred_dict[dataset])
+        cls._save_metadata(output_dir=output_dir, dataset_to_models=dataset_to_models)
+        return cls(dataset_to_models=dataset_to_models, output_dir=output_dir)
```

Comment:
True although it also depends, if you need to compute an error on all folds (which we do most of the times currently) then it will not save you in this case.